R/deploy-site.R

```diff
@@ -55,41 +55,76 @@ deploy_site_github <- function(
     stop("No repo detected, please supply one with `repo_slug`", call. = FALSE)
   }
 
-  # force execution before changing working directory
-  force(commit_message)
-
   rule("Deploying site", line = 2)
   rule("Installing package", line = 1)
   callr::rcmd("INSTALL", tarball, show = verbose, fail_on_status = TRUE)
 
-  build_site(pkg, ..., document = FALSE)
-
-  remote_url <- sprintf("git@github.com:%s.git", repo_slug)
-
   ssh_id_file <- "~/.ssh/id_rsa"
   rule("Setting up SSH id", line = 1)
   cat_line("Copying private key to: ", ssh_id_file)
   write_lines(rawToChar(openssl::base64_decode(ssh_id)), ssh_id_file)
   cat_line("Setting private key permissions to 0600")
   fs::file_chmod(ssh_id_file, "0600")
 
-  rule("Commiting built site", line = 1)
+  deploy_local(pkg, repo_slug = repo_slug, commit_message = commit_message)
+
+  rule("Deploy completed", line = 2)
+}
+
+deploy_local <- function(
+                         pkg = ".",
+                         repo_slug = NULL,
+                         commit_message = construct_commit_message(pkg)
+                         ) {
+  # force execution before changing working directory
```

Comment:
I am _pretty_ sure we can remove this now, or if not at least move the forcing to https://github.com/r-lib/pkgdown/pull/878/files#diff-d6cb0d4a4a9cc177286126fb35f6fea4R114

Reply:
Oh yeah, good call. I think we still need it because of lazy evaluation, but we can move it later.